### PR TITLE
Added dependancies for clustering and better structure

### PR DESCRIPTION
**Pull request by darrenlooby**

### CODE REVIEW

**Comment by darrenlooby** [comment]
Built basic structure of recursive configuration model and provided bare bones server get things going

**Comment by darrenlooby** [comment]
Example of the configuration that is presented to the app can be found (assuming host and port): http://localhost:3000/config

File: cluster/folk.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+app.use(helmet());
+app.use(cors());
+app.use(logger('dev'));
+
+// @TODO add favicon setting globally to the app
+//app.use(favicon(__dirname + '/favicon.ico'));
+
+// In dev mode - output the entire configuration to a server
+// @TODO put dev mode etc into envs
+
```
Comment (davidbebb) [inline cluster/folk.js:36]: Neat - definitely agree on switching this in env, but I like the interface for checking config

File: cluster/folk.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+app.get('/config', function (req, res) {
+  res.json(config);
+});
+
+// @TODO add secure end point
+app.listen(process.env.TERRA_PORT, function () {
+  console.log('Example app listening on port', process.env.TERRA_PORT);
+});
+
+console.log('\nAvailable Bower Packages\n', bowerPath(), '\n');
```
Comment (davidbebb) [inline cluster/folk.js]: Is this still needed?

File: cluster/master.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+tracer.warn('Cluster Master is running with', cpuCount, 'folks.');
+tracer.warn('env.LOCATION', process.env.LOCATION);
+
+// Create a worker for each CPU
+for (var i = 0; i < cpuCount; i += 1) {
+  // Build a object to pass to each folk allowing later access if required
+  var new_worker_env = {};
+  new_worker_env.WORKER_NAME = 'Worker_' + i;
+  // Run the folk
+  cluster.fork(new_worker_env);
```
Comment (davidbebb) [inline cluster/master.js:26]: Would be nice to also see a process pid for debug here

File: lib/buildConfig.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
+var cache = require('./cacheFiles.js');
+// Function to return the bower path by passing the bower package name
+var bowerPath = require('../../bower-discover');
```
Comment (davidbebb) [inline lib/buildConfig.js:3]: Is this correct? It looks like this path is outside the project dir?

File: package.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "terra",
+  "version": "1.0.0",
+  "description": "Node.js constructor of SPA projects.",
+  "main": "bower.js",
```
Comment (davidbebb) [inline package.json]: ?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/blankbox/terra.git"
+  },
+  "author": "Healthera <tech@healthera.co.uk> (https://github.com/Healthera)",
```
Comment (davidbebb) [inline package.json]: Nope :)

**Pull request merged**